src/compas_timber/ghpython/components/CT_BTLx_From_Params/code.py

```diff
@@ -0,0 +1,79 @@
+import inspect
+
+from compas.scene import Scene
+from ghpythonlib.componentbase import executingcomponent as component
+from Grasshopper.Kernel.GH_RuntimeMessageLevel import Warning
+
+from compas_timber.fabrication import BTLxProcessing
+from compas_timber.ghpython.ghcomponent_helpers import get_leaf_subclasses
+from compas_timber.ghpython.ghcomponent_helpers import manage_dynamic_params
+from compas_timber.ghpython.ghcomponent_helpers import rename_gh_output
+
+
+class BTLxFromParams(component):
+    def __init__(self):
+        super(BTLxFromParams, self).__init__()
+        self.classes = {}
+        for cls in get_leaf_subclasses(BTLxProcessing):
+            self.classes[cls.__name__] = cls
+        if ghenv.Component.Params.Output[0].NickName == "Processing":
+            self.processing_type = None
+        else:
+            self.processing_type = self.classes.get(ghenv.Component.Params.Output[0].NickName, None)
+
+    def RunScript(self, element, ref_side=0, *args):
```

Comment:
the default ref side here doesn't seem to work, I'm getting a nasty error when not setting a ref_side